File: src/main/java/zoo/api/service/info/InfoService.java
package zoo.api.service.info;

import zoo.api.bo.info.Info;

import java.util.List;

public interface InfoService {
    List<Info> findAllInfos();

    Info findInfoById(int id);

    Info addInfo(Info info);

    Info updateInfo(Info info);

    void deleteInfo(Info info);
}

File: zoo-ui/src/services/InfoService.js
import axios from 'axios';

const INFO_API_BASE_URL = "http://localhost:8080/zoo-api/infos/";

class InfoService {

    getAllInfos(){
        return axios.get(INFO_API_BASE_URL);
    }

    createInfo(info){
        return axios.post(INFO_API_BASE_URL, info);
    }

    getInfoById(infoId){
        return axios.get(INFO_API_BASE_URL + infoId);
    }

    updateInfo(info, infoId){
        return axios.put(INFO_API_BASE_URL + info, infoId);
    }

    deleteInfo(infoId){
        return axios.delete(INFO_API_BASE_URL + infoId);
    }
}

export default new InfoService()
